.github/workflows/cmake.yml

```diff
@@ -1,128 +1,77 @@
-            target: linux-arm
+            preset: linux-arm
 
           - os: ubuntu-20.04
-            container: ubuntu:18.04
+            container: openziti/ziti-builder
             name: Linux arm64
-            install: $common_ubuntu_deps crossbuild-essential-arm64
-            # build current openssl libs instead of using old versions provided by apt on ubuntu 18
-            cmake_opts: "`-DVCPKG_OVERLAY_PORTS=./vcpkg-overlays/linux-syslibs/ubuntu18`"
-            target: linux-arm64
+            preset: linux-arm64
 
     steps:
-      - name: install tools
-        if: ${{ matrix.install != null }}
-        run: |
-          apt -y update
-          apt -y install ${{ matrix.install }}
-
-      - name: install contemporary git
-        if: ${{ matrix.container != null }}
-        run: |
-          apt -y update
-          apt -y install software-properties-common
-          add-apt-repository -y ppa:git-core/ppa
-          apt -y update
-          apt -y install git
-          git config --global --add safe.directory $GITHUB_WORKSPACE
-          git --version
-
-      # run this step if Linux and arm or arm64 
-      - name: subscribe to apt multiarch ports and install libssl-dev for target arch
-        if: matrix.container != null && matrix.target != 'linux-x64'
-        run: |
-          case ${{ matrix.target }} in
-            linux-arm)   TARGETARCH=armhf;;
-            linux-arm64) TARGETARCH=arm64;;
-          esac
-          dpkg --add-architecture ${TARGETARCH}
-          cat <<EOF > /etc/apt/sources.list.d/crossbuild.list
-          deb [arch=${TARGETARCH}] http://ports.ubuntu.com/ bionic main restricted
-          deb [arch=${TARGETARCH}] http://ports.ubuntu.com/ bionic-updates main restricted
-          deb [arch=${TARGETARCH}] http://ports.ubuntu.com/ bionic universe
-          deb [arch=${TARGETARCH}] http://ports.ubuntu.com/ bionic-updates universe
-          deb [arch=${TARGETARCH}] http://ports.ubuntu.com/ bionic multiverse
-          deb [arch=${TARGETARCH}] http://ports.ubuntu.com/ bionic-updates multiverse
-          deb [arch=${TARGETARCH}] http://ports.ubuntu.com/ bionic-backports main restricted universe multiverse
-          EOF
-          sed -Ei 's/^deb/deb [arch=amd64]/g' /etc/apt/sources.list
-          apt-get update
-          apt-get -y install libssl-dev:${TARGETARCH} zlib1g-dev:${TARGETARCH}
-
       - name: checkout workspace
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
-      - name: install contemporary cmake
-        uses: lukka/get-cmake@latest
-
-      - uses: lukka/run-vcpkg@v10
-        with:
-          # use 2023.02.24 vcpkg baseline,
-          # see https://learn.microsoft.com/en-us/vcpkg/users/examples/versioning.getting-started#builtin-baseline
-          vcpkgGitCommitId: 'a7b6122f6b6504d16d96117336a0562693579933'
+      - name: Run CMake for Linux
+        if: ${{ matrix.container != null }}
+        run: |
+          ./ziti-builder.sh \
```

Comment:
by using the script you're not taking advantage of vcpkg caching which reduces the build times considerably
also, it is not clear what VCPKG baseline/commit the script would be using

Reply:
Re: binary caching for vcpkg

Could you let me know if you meant to suggest a persistent cache for GH Actions runs? If so, is that working for the zitify or ziti-sdk-c projects? I understand the hosted job runners start with a pristine filesystem, so it's necessary to self-host the runner to make other storage available. That's doable but isn't expedient unless build minutes become a scarce resource. I've used the artifact upload/download Actions to move files around, but not a cache directory.

You might have meant that a developer running the script locally doesn't benefit from their VCPKG binary cache. That's something I hadn't considered. If the cache is in a predictable file path, we could make it available inside the builder container easily. Is that what you had in mind?

---

Re: VCPKG baseline/commit

Will you please clarify this one? The VCPKG preset file is from the same commit as the script because they're part of the repo. I suspect you meant something else.

Reply:
GH workflows come with a way of caching _stuff_ between runs. Usually actions maintain their own caches.
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows

For example `run-vcpkg` will restore 3rd party binaries from previous runs so that the current run does not have rebuild everything 

when you build locally, your VCPKG_ROOT serves as the cache (even if you clean your build directory 3rd party dependencies are repopulated from VCPKG_ROOT/packages)

Reply:
VCPKG baseline refers to git commit id for vcpkg repo. Essentially if fixes the versions of third party dependencies that we use.

Reply:
here is an example of cache reuse in ziti-sdk-c https://github.com/openziti/ziti-sdk-c/actions/runs/5270170584/jobs/9529313420#step:5:83

Reply:
> it is not clear what VCPKG baseline/commit the script would be using

@ekoby If I understand correctly, the baseline you're referring to is defined in the `ziti-builder` container image [on this line](https://github.com/openziti/ziti-builder/blob/main/docker-image/Dockerfile#L69). Is it sufficient to pin the VCPKG baseline in this way, or do you see a need to configure the VCPK version in the workflow runs that consume the `ziti-builder` image?